### PR TITLE
Dpl 211

### PR DESCRIPTION
**Pull request by emrojo**
Closes #

Changes proposed in this pull request:

*
*
* ...

### CODE REVIEW

File: app/models/accessionable/base.rb [hunk 1]
```diff
@@ -79,16 +79,82 @@ def label_scope
...
+        name == :country_of_origin
+      end
+    end
+
+    # Value serialization for collection date in accessioning XML generation
+    # It will return a valid collection date or 'not collected' if nothing provided or invalid
+    # It also allow other config settings for the XML service like the list defined inside OTHER_DEFAULT_SETTINGS
+    # NB: this regexp is defined in <https://www.ebi.ac.uk/ena/browser/api/xml/ERC000011>
+    class FieldCollectionDate < FieldSerializer
+      # rubocop:disable Layout/LineLength
```
Comment (stevieing) [inline app/models/accessionable/base.rb:123]: Is this a regexp to check for a valid date?

**Review by stevieing — APPROVED** [review]

File: app/models/accessionable/base.rb [hunk 2]
```diff
@@ -79,16 +79,82 @@ def label_scope
...
 
+    # Value serialization for accessioning XML generation
+    # It will return the same value without any changes and it applies
+    # to all tags
+    class FieldSerializer
+      NOT_COLLECTED = 'not collected'
+      NOT_PROVIDED = 'not provided'
+      RESTRICTED_ACCESS = 'restricted access'
+
+      OTHER_DEFAULT_SETTINGS = [NOT_COLLECTED, NOT_PROVIDED, RESTRICTED_ACCESS].freeze
```
Comment (yoldas) [inline app/models/accessionable/base.rb:90]: 'not applicable' is also given in config/locales/metadata/en.yml  country_of_origin help

File: lib/accession/accession/tag.rb [hunk 3]
```diff
@@ -58,5 +58,53 @@ def array_express_label
     def attributes
       %i[services value name groups ebi_name]
     end
+
+    # Some helper methods for displaying information from tags
+    module HelperTagValue
+      NOT_COLLECTED = 'not collected'
+      NOT_PROVIDED = 'not provided'
+      RESTRICTED_ACCESS = 'restricted access'
+      OTHER_DEFAULT_SETTINGS = [NOT_COLLECTED, NOT_PROVIDED, RESTRICTED_ACCESS].freeze
```
Comment (yoldas) [inline lib/accession/accession/tag.rb:67]: Check if 'not applicable' is necessary. https://ena-docs.readthedocs.io/en/latest/submit/samples/missing-values.html

File: lib/accession/accession/tag.rb [hunk 4]
```diff
@@ -58,5 +58,53 @@ def array_express_label
...
+  end
+
+  # Value serialization for collection date in accessioning XML generation
+  # It will return a valid collection date or 'not collected' if nothing provided or invalid
+  # It also allow other config settings for the XML service like the list defined inside OTHER_DEFAULT_SETTINGS
+  # NB: this regexp is defined in <https://www.ebi.ac.uk/ena/browser/api/xml/ERC000011>
+  class TagCollectionDate < Tag
+    # rubocop:disable Layout/LineLength
+    REGEXP = %r{(^[12][0-9]{3}(-(0[1-9]|1[0-2])(-(0[1-9]|[12][0-9]|3[01])(T[0-9]{2}:[0-9]{2}(:[0-9]{2})?Z?([+-][0-9]{1,2})?)?)?)?(/[0-9]{4}(-[0-9]{2}(-[0-9]{2}(T[0-9]{2}:[0-9]{2}(:[0-9]{2})?Z?([+-][0-9]{1,2})?)?)?)?)?$)}
+      .freeze
```
Comment (yoldas) [inline lib/accession/accession/tag.rb:99]: Does this accept YYYY-MM and YYYY formats ?
Does this need to accept " Missing Value Reporting Terms"  ?

File: spec/lib/accession/sample_spec.rb [hunk 5]
```diff
@@ -145,4 +160,70 @@
...
+        expect(sample.to_xml).to include(COUNTRY_TAG)
+        expect(find_value_at_tag(sample.to_xml, COUNTRY_TAG)).to eq('not collected')
+      end
+    end
+
+    context 'with collection date' do
+      it 'includes collection date' do
+        sample = described_class.new(tag_list, create(:sample_for_accessioning_with_open_study))
+        expect(sample.to_xml).to include(COLLECTION_DATE_TAG)
+        expect(find_value_at_tag(sample.to_xml, COLLECTION_DATE_TAG)).to eq('2000-01-01T00:00')
```
Comment (yoldas) [inline spec/lib/accession/sample_spec.rb:201]: Do the date-times need to have ISO separator 'T' ?
Can date-times have missing minutes part?
Can date-times have date and a trailing T only, for example '2000-01-01T' ?

File: app/models/accessionable/base.rb (hunk 1 above)
Comment (emrojo) [reply app/models/accessionable/base.rb:123]: Yes, to me it seems it has been automatically generated. It is what they specified in https://www.ebi.ac.uk/ena/browser/view/ERC000011

**Pull request merged**